awd/listen.py

```diff
@@ -58,9 +79,20 @@ def listen(sqs_base_url, sqs_output_queue, log_source_email, log_recipient_email
             if "'ResultType': 'error'" in message["Body"]:
                 logger.error(f'DOI: {doi}, Result: {message.get("Body")}')
                 sqs.delete(sqs_base_url, sqs_output_queue, message["ReceiptHandle"])
+                if dynamodb_client.retry_attempts_exceeded(
+                    doi_table, doi, retry_threshold
+                ):
+                    dynamodb_client.update_doi_item_status_in_database(
+                        doi_table, doi, 4
```

Comment:
It'd clearer and more robust to call the function with an enum (that you have created), instead of its explicit value. So something like Status.Processing or Status.Failed, instead of 1 or 2.